engine/src/main/scala/cromwell/webservice/WorkflowQueryPagination.scala

```diff
@@ -0,0 +1,41 @@
+package cromwell.webservice
+
+import cromwell.services.MetadataServiceActor.QueryMetadata
+import spray.http.HttpHeaders.Link
+import spray.http.{HttpHeader, Uri}
+
+
+trait WorkflowQueryPagination {
+
+  protected def generatePaginationParams(page: Int, pageSize: Int): String = {
+    s"page=$page&pagesize=$pageSize"
+  }
+
+  //Generates link headers for pagination navigation https://tools.ietf.org/html/rfc5988#page-6
+  protected def generateLinkHeaders(uri: Uri, metadata: Option[QueryMetadata]): Seq[HttpHeader] = {
+    //strip off the query params
+    val baseUrl = uri.scheme + ":" + uri.authority + uri.path
+    metadata match {
+      case Some(meta) =>
+        (meta.page, meta.pageSize) match {
+          case (Some(p), Some(ps)) =>
+
+            val firstLink = Link(Uri(baseUrl).withQuery(generatePaginationParams(1, ps)), Link.first)
```

Comment:
This won't work perfectly for us either on hotfix -- we may have to submit more patches there. First, we have to deploy viewscreen (the cromwell consumer) so we can get a better idea of what we need to make it work.

I'm also not sure what headers should return from a POST request - personally, I don't think Link headers should be in the response to POST requests. I think a metadata object should be included in the response which could be used by the client to construct paginated POST requests. 

And if you're doing that for POST requests, it could make sense to remove Link header pagination and just support this metadata object for GET and POST requests. This implementation probably requires a bit more discussion though.